core/rawdb/accessors_account.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package rawdb
+
+import (
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/common/dbutils"
+	"github.com/ledgerwatch/turbo-geth/core/types/accounts"
+)
+
+// ReadAccount retrieves the version number of the database.
+func ReadAccount(db DatabaseReader, addrHash common.Hash, acc *accounts.Account) (bool, error) {
+	addrHashBytes := addrHash[:]
+	enc, err := db.Get(dbutils.AccountsBucket, addrHashBytes)
+	if err != nil {
+		return false, err
+	}
+	root, err := db.Get(dbutils.IntermediateTrieHashBucket, addrHashBytes)
```

Comment:
It is not always guaranteed that such entry exists. What if part of the storage trie is loaded into memory and therefore not present in the IntermediateHashBucket?